### PR TITLE
#37 singular product page feature and spec section

### PR DESCRIPTION
**Pull request by camargov**

### CODE REVIEW

File: src/components/singular-product/singular-product.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
+import React from "react"
```
Comment (samgildea) [inline src/components/singular-product/singular-product.js]: This looks great so far!  Can you add this to the product template page in "/templates/product.js" and ultimately remove the singular product page javascript file as it is only necessary for generated product pages.

File: src/components/singular-product/singular-product-styles.js [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+  margin-top: 100px;
+`
+
+export const FeatureIcon = styled.div`
+  width: 3vw;
+  height: 45px;
+`
+
+export const FeatureSpecHeading = styled.div`
+  font-size: calc(40px + (40 - 32) * ((100vw - 300px) / (1440 - 300)));
```
Comment (samgildea) [inline src/components/singular-product/singular-product-styles.js]: Use the pre defined typography styles.

File: src/components/singular-product/singular-product-styles.js [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    padding-top: 54px;
+  }
+`
+
+export const Icon = styled.div``
+
+export const IconImage = styled.div`
+  width: 4vw;
+  height: 55px;
+  background-color: #C4C4C4;
```
Comment (samgildea) [inline src/components/singular-product/singular-product-styles.js]: Use colors from colors.js

File: src/components/singular-product/singular-product-styles.js [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    padding-top: 45px;
+  }
+`
+
+export const ImageContainer = styled.div``
+
+export const MainImage = styled.div`
+  width: 47vw;
+  height: 432px;
+  background-color: #C4C4C4;
```
Comment (samgildea) [inline src/components/singular-product/singular-product-styles.js]: Use colors from colors.js

File: src/components/singular-product/singular-product-styles.js [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+    justify-content: center;
+  }
+`
+
+export const SmallImage = styled.div`
+  width: 4vw;
+  height: 56px;
+  margin-right: 18px;
+  margin-top: 44px;
+  background-color: #C4C4C4;
```
Comment (samgildea) [inline src/components/singular-product/singular-product-styles.js]: Use colors from colors.js

File: src/components/singular-product/singular-product-styles.js [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+  padding-top: 90px;
+
+  @media (min-width: ${dimensions.maxwidthTablet}px) {
+    display: flex;
+  }
+`
+
+export const FeatureName = styled.div`
+  text-transform: uppercase;
+  font-size: calc(20px + (20 - 16) * ((100vw - 300px) / (1440 - 300)));
```
Comment (samgildea) [inline src/components/singular-product/singular-product-styles.js]: Use the pre defined typography styles.

File: src/components/singular-product/singular-product-styles.js [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+  padding-top: 49px;
+
+  @media (max-width: ${dimensions.maxwidthTablet}px) {
+    padding-top: 63px;
+  }
+`
+
+export const ProductName = styled.div`
+  font-weight: bold;
+  font-size: calc(50px + (50 - 32) * ((100vw - 300px) / (1440 - 300)));
```
Comment (samgildea) [inline src/components/singular-product/singular-product-styles.js]: Use the pre defined typography styles.

File: src/components/singular-product/singular-product-styles.js [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+  padding-top: 27px;
+
+  @media (max-width: ${dimensions.maxwidthTablet}px) {
+    
+  }
+`
+
+export const ProductDescription = styled.div`
+  font-size: calc(18px + (18 - 16) * ((100vw - 300px) / (1440 - 300)));
```
Comment (samgildea) [inline src/components/singular-product/singular-product-styles.js]: Use the pre defined typography styles.

**Review by samgildea — CHANGES_REQUESTED** [review]

File: src/components/singular-product/singular-product.js [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+              {data.features.map((feature, id) => {
+                return (
+                  <Feature>
+                    <FeatureName>{feature.feature_name}</FeatureName>
+                    <FeatureImage>
+                      <img key={id} 
+                      className={id === 0? "first-image" : ""} 
+                      className={id !== 0? "not-image" : ""}
+                      src={feature.feature_image.url} 
+                      alt="Feature of a product"/>
```
Comment (josephmannis) [inline src/components/singular-product/singular-product.js]: Can this alt come from the CMS please

File: src/components/singular-product/singular-product.js [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+          </IconSection>
+          <FeatureSpecSection>
+            <FeatureIcon>
+              <img src={data.feature_icon.url} alt="Orange checklist icon"/>
+            </FeatureIcon>
+            <FeatureSpecHeading>{data.feature_title}</FeatureSpecHeading>
+            <FeatureSpecFeatures>
+              {data.features.map((feature, id) => {
+                return (
+                  <Feature>
```
Comment (josephmannis) [inline src/components/singular-product/singular-product.js]: The key should be in the top-level component

File: src/templates/product.js [hunk 11]
```diff
@@ -1,20 +1,119 @@
...
+          <SmallImageContainer>
+            <SmallImage></SmallImage>
+            <SmallImage></SmallImage>
+            <SmallImage></SmallImage>
+            <SmallImage></SmallImage>
+            <SmallImage></SmallImage>
+          </SmallImageContainer>
+        </ImageContainer>
+        <TextContainer>
+          <ProductType>{products.data.product_type}</ProductType>
```
Comment (josephmannis) [inline src/templates/product.js]: Are these changes from the other PR? https://github.com/Scout-NU/powerhouse-site/pull/48#pullrequestreview-638265192 

If so, can you make sure you don't have stuff from other PRs in another in the future?

File: src/templates/product.js [hunk 12]
```diff
@@ -1,20 +1,119 @@
...
+          <ProductName>{products.data.product_title}</ProductName>
+          <ProductDescription>
+            {products.data.product_description.text}
+          </ProductDescription>
+        </TextContainer>
+      </HeroSection>
+      <IconSection>
+        <IconSectionHeading>{products.data.icon_section_title}</IconSectionHeading>
+        <IconContainer>
+          {products.data.icons.map(icon => {
```
Comment (josephmannis) [inline src/templates/product.js]: Keys!

File: src/templates/product.js [hunk 13]
```diff
@@ -1,20 +1,119 @@
...
+                <IconCaption>{icon.icon_text}</IconCaption>
+              </Icon>
+            )
+          })}
+        </IconContainer>
+      </IconSection>
+
+      <FeatureSpecSection>
+        <FeatureIcon>
+          <img src={products.data.feature_icon.url} alt="Orange checklist icon" />
```
Comment (josephmannis) [inline src/templates/product.js]: CMS alt please

File: src/templates/product.js [hunk 14]
```diff
@@ -1,20 +1,119 @@
...
+        </FeatureIcon>
+        <FeatureSpecHeading>{products.data.feature_title}</FeatureSpecHeading>
+        <FeatureSpecFeatures>
+          {products.data.features.map((feature, id) => {
+            return (
+              <Feature>
```
Comment (josephmannis) [inline src/templates/product.js]: Keys at top level please

File: src/templates/product.js [hunk 15]
```diff
@@ -1,20 +1,119 @@
...
+                <FeatureName>{feature.feature_name}</FeatureName>
+                <FeatureImage>
+                  <img
+                    key={id}
+                    className={id === 0 ? "first-image" : ""}
+                    className={id !== 0 ? "not-image" : ""}
+                    src={feature.feature_image.url}
+                    alt="Feature of a product"
```
Comment (josephmannis) [inline src/templates/product.js]: CMS alt text please

File: src/components/singular-product/singular-product.js [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+                <ProductDescription>{data.product_description.text}</ProductDescription>
+              </TextContainer>
+          </HeroSection>
+          <IconSection>
+            <IconSectionHeading>{data.icon_section_title}</IconSectionHeading>
+            <IconContainer>
+              {data.icons.map(icon => {
+                return (
+                  <Icon>
+                    <IconImage></IconImage>
```
Comment (josephmannis) [inline src/components/singular-product/singular-product.js]: Are these supposed to contain something?

**Review by josephmannis — CHANGES_REQUESTED** [review]
Few things!

File: src/components/about/about.js [hunk 17]
```diff
@@ -0,0 +1,197 @@
...
+export default function About({ data }) {
+  return (
+    <>
+      <AboutHeaderText>
+        <H2>About</H2>
+      </AboutHeaderText>
+
+      <MissionSection>
+        <MissionImage>
+          <img src={data.mission_image.url} />
```
Comment (josephmannis) [inline src/components/about/about.js]: Can you add the alt text from the CMS please?

File: src/components/about/about.js [hunk 18]
```diff
@@ -0,0 +1,197 @@
...
+        </MissionImage>
+
+        <MissionText>
+          <IconPlaceholder>
+            <img src={data.mission_icon.url} />
```
Comment (josephmannis) [inline src/components/about/about.js]: Same here

File: src/components/about/about.js [hunk 19]
```diff
@@ -0,0 +1,197 @@
...
+          <H2>{data.mission_heading}</H2>
+          <MissionTextParagraph>
+            <Body>{data.mission_description}</Body>
+          </MissionTextParagraph>
+        </MissionText>
+      </MissionSection>
+
+      <IndustryContextSection>
+        <IndustryContextBackgroundImage>
+          <img src={data.industry_context_background_image.url} />
```
Comment (josephmannis) [inline src/components/about/about.js]: And here

File: src/components/about/about.js [hunk 20]
```diff
@@ -0,0 +1,197 @@
...
+        </IndustryContextBackgroundImage>
+        <IndustryContextContainer>
+          <div>
+            <IconPlaceholder>
+              <img src={data.industry_context_icon.url} />
```
Comment (josephmannis) [inline src/components/about/about.js]: Here too

File: src/components/about/about.js [hunk 21]
```diff
@@ -0,0 +1,197 @@
...
+            </IconPlaceholder>
+            <H2>{data.industry_context_heading}</H2>
+            <IndustryContextExamples>
+              {data.industry_context_explanation.map((example, id) => {
+                return (
+                  <IndustryContextGroup key={id}>
+                    <IndustryContextImage>
+                      <img src={example.explanation_image.url} />
```
Comment (josephmannis) [inline src/components/about/about.js]: Here

File: src/components/about/about.js [hunk 22]
```diff
@@ -0,0 +1,197 @@
...
+      <div>
+        <ProductBenefitHeader>
+          <H2>{data.company_comparison_section_heading}</H2>
+        </ProductBenefitHeader>
+        <ProductBenefitsContainer>
+          {data.product_benefit.map((benefit, id) => {
+            return (
+              <BenefitGroup key={id}>
+                <BenefitImage>
+                  <img src={benefit.product_benefit_image.url} />
```
Comment (josephmannis) [inline src/components/about/about.js]: Alt

File: src/components/about/about.js [hunk 23]
```diff
@@ -0,0 +1,197 @@
...
+        </ProductBenefitsContainer>
+      </div>
+
+      <CompanyComparisonSection>
+        <div>
+          <table>
+            <tr>
+              <th></th>
+              {data.company.map((competitor, id) => (
+                <>
```
Comment (josephmannis) [inline src/components/about/about.js]: Can you remove this fragment?

File: src/components/about/about.js [hunk 24]
```diff
@@ -0,0 +1,197 @@
...
+        <CompanyTeamSection>
+          <CompanyTeamHeading>
+            <H2>{data.company_team_heading}</H2>
+          </CompanyTeamHeading>
+          <CompanyTeamMemberGroup>
+            {data.team_member_information.map((member, id) => {
+              return (
+                <TeamMember key={id}>
+                  <MemberImage>
+                    <img src={member.team_member_image.url} />
```
Comment (josephmannis) [inline src/components/about/about.js]: Alt

File: src/templates/product.js [hunk 25]
```diff
@@ -1,36 +1,198 @@
...
   const products = data.allPrismicProduct.edges[0].node
+  const [mainImage, setMainImage] = useState(0)
 
-  return <h1>{products.data.product_title}</h1>
+  return (
+    <Layout>
+      <SingularProductContainer>
+        <HeroSection>
+          <ImageContainer>
+            <img
```
Comment (josephmannis) [inline src/templates/product.js:41]: Alt

File: src/templates/product.js [hunk 26]
```diff
@@ -1,36 +1,198 @@
...
+            {products.data.product_images.map((image, id) => {
+              return (
+                <>
+                  <img
+                    onClick={() => setMainImage(id)}
+                    src={image.image.url}
+                    alt={image.alt}
+                    key={id}
+                  />
+                  {console.log(image)}
```
Comment (josephmannis) [inline src/templates/product.js]: Can you remove this and the fragment please?

File: src/templates/product.js [hunk 27]
```diff
@@ -1,36 +1,198 @@
...
+            <img
+              src={products.data.feature_icon.url}
+              alt={products.data.feature_icon.alt}
+            />
+          </FeatureIcon>
+          <H2>{products.data.feature_title}</H2>
+          <div>
+            {products.data.features.map((feature, id) => {
+              return (
+                <Feature>
```
Comment (josephmannis) [inline src/templates/product.js]: Need a key

**Review by josephmannis — CHANGES_REQUESTED** [review]
Just a few little things

**Pull request merged**